docs/source/about_arrow.md

```diff
@@ -0,0 +1,35 @@
+# Datasets 🤝 Arrow
+
+TO DO:
+
+A brief introduction on why Datasets chose to use Arrow. For example, include some context and background like design decisions and constraints. The user should understand why we decided to use Arrow instead of something else.
```

Comment:
TODO: Brief intro on why Datasets chose to use Arrow, provide some historical context and background.